client/src/components/composite/Profile/ProfileEdit/ProfileEdit.tsx

```diff
@@ -0,0 +1,95 @@
+import { ReducedUserAdditionalInfo } from "models/User"
+import TextInput from "components/generic/TextInputComponent/TextInput"
+import Button from "components/generic/FigmaButtons/FigmaButton"
+import CloseButton from "assets/icons/x.svg?react"
+import { useState } from "react"
+
+interface IProfileEdit<T extends Partial<ReducedUserAdditionalInfo>> {
+  title: string
+  fields: {
+    fieldName: keyof T
+    defaultFieldValue: string
+  }[]
+  onEdit: (fields: Partial<T>) => void
+}
+
+const nameTransformer = (
+  originalName: keyof ReducedUserAdditionalInfo
+): string => {
+  switch (originalName) {
+    case "date_of_birth":
+      return "Date of Birth"
+    case "does_snowboarding":
+      return "Does Snowboarding"
+    case "does_racing":
+      return "Does Racing"
+    case "does_ski":
+      return "Does Ski"
+    case "phone_number":
+      return "Mobile number"
+    case "gender":
+      return "Gender"
+    case "emergency_contact":
+      return "Emergency contact info"
+    case "first_name":
+      return "First Name"
+    case "last_name":
+      return "Last Name"
+    case "dietary_requirements":
+      return "Dietary Requirements"
+    case "ethnicity":
+      return "Ethnicity"
+    case "faculty":
+      return "Faculty"
+    case "university":
+      return "University"
+    case "student_id":
+      return "UoA Student ID Number"
+    case "university_year":
+      return "University Year"
+  }
+}
+
+const ProfileEdit = <T extends Partial<ReducedUserAdditionalInfo>>({
+  title,
+  fields
+}: IProfileEdit<T>) => {
+  function handleClose() {
+    setClosed(true)
+  }
+  const [closed, setClosed] = useState(false)
+  if (!closed) {
```

Comment:
dont handle it in here, should be controlled outside the component (can just pass in an `onClose` and `isOpen`)